src/Features/Core/Portable/StackTraceExplorer/IStackTraceExplorerService.cs

```diff
@@ -2,25 +2,28 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.FindUsages;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Remote;
 
 namespace Microsoft.CodeAnalysis.StackTraceExplorer
 {
-    internal interface IStackTraceExplorerService : ILanguageService
+    internal interface IStackTraceExplorerService : IWorkspaceService
     {
         /// <summary>
-        /// Given the <see cref="ParsedStackFrame.TypeSpan"/>, get the equivalent name
-        /// in metadata that can be used to look up the type
+        /// If the <paramref name="frame"/> has file information, attempts to map it to existing documents
+        /// in a solution. Looks for an exact filepath match first, then defaults to 
+        /// a best guess.
         /// </summary>
-        string GetTypeMetadataName(string className);
+        (Document? document, int line) GetDocumentAndLine(Solution solution, ParsedFrame frame);
```

Comment:
we have a type called DocumentSpan that would likely be appropriate here.

Reply:
I'm not fully convinced, mostly because it would still be mapping from span to line as an exercise of the caller. It's not terribly hard, but still tedious. And then it's unclear... is the span the line? The symbol definition location?